### PR TITLE
Support for send documents and fix previous change that broke post (issue #8)

### PR DESCRIPTION
**Pull request by frangeris**
Adding support for send documents, Fixing bug with Request Options and guzzle, body and form_params, [issue #8](https://github.com/99designs/rightsignature-php/issues/8)

### CODE REVIEW

**Comment by harto** [comment]
Thanks! It looks like there might be some formatting inconsistencies (tabs instead of spaces etc).
I introduced `php-cs-fixer` in #6. Could you run it across the codebase and commit the changes?
This might work:
```
$ vendor/bin/php-cs-fixer fix .
```

**Comment by frangeris** [comment]
That's rare, I'm using that vendor for PSR-2 format, the only change that it detects when I run again is in [this line](https://github.com/frangeris/rightsignature-php/blob/master/classes/RightSignature/HttpClient.php#L91), is that what you mean?

**Comment by harto** [comment]
Oh — weird. I was thinking about `DocumentTest`. If you look at the diff in this PR you can see the indentation is different in that file.

**Comment by frangeris** [comment]
@harto I was using tabs indent 4, changed to 2, please verify now...

**Comment by harto** [comment]
@frangeris it should be 4 spaces, not tabs.
Do you think I should add an `.editorconfig` or similar?

**Comment by frangeris** [comment]
@harto, sorry my bad, I refresh the file, please verify, if still, please add `.editorconfig` file and so avoids these issues...

File: tests/RightSignature/DocumentTest.php [hunk 1]
```diff
@@ -109,4 +109,46 @@ public function testDocumentDetails()
...
+                ],
+            ],
+        ];
+
+        $client = \Mockery::mock('client');
+        $client->shouldReceive('post')
+                ->withAnyArgs()
+                ->andReturn($response);
+
+        $file = fopen('sample-file.pdf', 'w');
```
Comment (harto) [inline tests/RightSignature/DocumentTest.php]: Perhaps you could use http://php.net/manual/en/function.tmpfile.php to create the input document. That way the file doesn't hang around the repo if the test crashes before `unlink` gets called.

**Comment by harto** [comment]
Looks good @frangeris. Thanks :)

**Comment by frangeris** [comment]
Hi @harto, I'm waiting for this feature... anything else before merge?

Thank you...

**Comment by harto** [comment]
Oops, I forgot. Sorry :)

**Comment by harto** [comment]
Tagged `2.0.0-RC3`

**Pull request merged**